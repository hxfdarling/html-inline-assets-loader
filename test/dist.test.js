
import 'jest-extended';

const { compile, evaluated } = require('./helpers');

describe('dist loader', () => {
  it('在development模式下不引入', async () => {
    const stats = await compile('dist.html');

    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });
  it('在production模式下引入', async () => {
    process.env.NODE_ENV = 'production';
    const stats = await compile('dist.html');

    expect(stats.compilation.warnings).toBeArrayOfSize(0);
    expect(stats.compilation.errors).toBeArrayOfSize(0);

    const { modules } = stats.toJson();
    const [module] = modules;
    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules))
      .toMatchSnapshot('module (evaluated)');
  });
});
