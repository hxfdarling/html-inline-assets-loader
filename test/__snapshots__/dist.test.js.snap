// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dist loader 在development模式下不引入: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <title>imt application</title>
    
    
  </head>
  <body>

</body></html>"
`;

exports[`dist loader 在development模式下不引入: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <title>imt application</title>\\\\n    \\\\n    \\\\n  </head>\\\\n  <body>\\\\n\\\\n</body></html>\\";"`;

exports[`dist loader 在production模式下引入: module (evaluated) 1`] = `
"<!DOCTYPE html><html><head>
    <title>imt application</title>
    <script>\\"use strict\\";

console.log('inline');</script>
    <script src=\\"inline_fde1ab.js\\"></script>
  </head>
  <body>

</body></html>"
`;

exports[`dist loader 在production模式下引入: module 1`] = `"module.exports = \\"<!DOCTYPE html><html><head>\\\\n    <title>imt application</title>\\\\n    <script>\\\\\\"use strict\\\\\\";\\\\n\\\\nconsole.log('inline');</script>\\\\n    <script src=\\\\\\"inline_fde1ab.js\\\\\\"></script>\\\\n  </head>\\\\n  <body>\\\\n\\\\n</body></html>\\";"`;
